calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/camera_calibrator.py

```diff
@@ -795,6 +874,23 @@ def process_evaluation_results(
     def on_consumed(self):
         self.data_source.consumed()
 
+    def save_parameters(self, filename):
+        data_coll_params = self.data_collector.get_parameters_values()
+        board_params = self.board_parameters.get_parameters_values()
+        detector_params = self.detector.get_parameters_values()
+        calibrator_type = self.calibrator_type_combobox.currentData()
+        print(calibrator_type.value["name"], flush=True)
+        calib_params = self.calibrator_dict[calibrator_type].get_parameters_values()
+        with open(filename, "w") as file:
+            yaml.dump({"board_parameters": board_params}, file, default_flow_style=False)
+            yaml.dump({"board_type": self.board_type.value["name"]}, file, default_flow_style=False)
+            yaml.dump(
+                {"calibrator_type": calibrator_type.value["name"]}, file, default_flow_style=False
+            )
+            yaml.dump({"calibration_parameters": calib_params}, file, default_flow_style=False)
+            yaml.dump({"data_collector": data_coll_params}, file, default_flow_style=False)
+            yaml.dump({"detector_params": detector_params}, file, default_flow_style=False)
```

Comment:
can you not dump all of these at once?

Reply:
what about doing something like this?

`all_params = {
    "board_parameters": board_params,
    "board_type" : self.board_type.value["name"],
    "calibrator_type": calibrator_type.value["name"],
    "calibration_parameters": calib_params,
    "data_collector": data_coll_params,
    "detector_params": detector_params
}

yaml.dump(all_params, file, default_flow_style=False)`

Reply:
Yes, that is what I meant

Reply:
f2ad5cce4cffc72284d0b575020c40851475872e